docker-compose.yml

```diff
@@ -0,0 +1,63 @@
+version: '3'
+
+services:
+    assets:
+        build:
+            context: .
+            dockerfile: Dockerfile.assets
+            args:
+                image_tag: ${IMAGE_TAG}
+        image: elifesciences/journal_assets:${IMAGE_TAG}
+        depends_on:
+            - composer
+            - npm
+    cli:
+        build:
+            context: .
+            dockerfile: Dockerfile.cli
+            args:
+                image_tag: ${IMAGE_TAG}
+        image: elifesciences/journal_cli:${IMAGE_TAG}
+        volumes:
+            - ./.docker/parameters.yml:/srv/journal/app/config/parameters.yml
+        depends_on:
+            - assets
+            - composer
+    composer:
+        build:
+            context: .
+            dockerfile: Dockerfile.composer
+            args:
+                composer_dev_arg: ${COMPOSER_DEV_ARG}
+        image: elifesciences/journal_composer:${IMAGE_TAG}
+        command: /bin/bash
+    fpm:
+        build:
+            context: .
+            dockerfile: Dockerfile.fpm
+            args:
+                image_tag: ${IMAGE_TAG}
+        image: elifesciences/journal_fpm:${IMAGE_TAG}
+        volumes:
+            - ./.docker/parameters.yml:/srv/journal/app/config/parameters.yml
+            - web:/srv/journal/web
+        depends_on:
+            - assets
+            - composer
+    npm:
+        build:
+            context: .
+            dockerfile: Dockerfile.npm
+        image: elifesciences/journal_npm:${IMAGE_TAG}
+    web:
+        image: nginx:1.13.7
+        ports:
+            - '8080:80'
+        volumes:
+            - ./.docker/nginx-default.conf:/etc/nginx/conf.d/default.conf
+            - web:/srv/journal/web
```

Comment:
can't track why the `web` folder is a volume, but in fact it can't be in general as:
- the image being used in production won't have such a volume, so needs to be able to work without one
- local development may mount the local `./web` folder to override `/srv/journal/web`, but not a stateful volume like this one

Reply:
`web` needs to be served through Nginx as it contains assets. Alternative is a separate Dockerfile and a `COPY`.

Reply:
Considering assets are versioned with the rest of the code, they need to be published as an image; this is not necessary to run `ci`, but it will be later when the images are used by `end2end` and all other environments. With some orchestration it may not have to be an `nginx` image as it could be used to initialize a volume; but it's probably easier to use `nginx` without HTTPS or other frills and have the production nginx forward to this one.